### ISSUE TITLE
`eth_getTransactionByHash` returns null

### ISSUE THREAD

**Issue by Rjected** [paradigmxyz/reth#1906]
### Describe the bug

The hive test for `eth_getTransactionByHash` fails, returning `null` instead of the transaction.

```diff
>>  {"jsonrpc":"2.0","id":1,"method":"eth_getTransactionByHash","params":["0x0d9ba049a158972e7fc1066122ceb31e431483ebf84f90f845f02e326942d467"]}
<<  {"jsonrpc":"2.0","result":null,"id":1}
response differs from expected:
 {
   "id": 1,
   "jsonrpc": "2.0",
-  "result": null
+  "result": {
+    "blockHash": "0xfe21bb173f43067a9f90cfc59bbb6830a7a2929b5de4a61f372a9db28e87f9ae",
+    "blockNumber": "0x2",
+    "chainId": "0x539",
+    "from": "0x658bdf435d810c91414ec09147daa6db62406379",
+    "gas": "0x5208",
+    "gasPrice": "0x2db08787",
+    "hash": "0x0d9ba049a158972e7fc1066122ceb31e431483ebf84f90f845f02e326942d467",
+    "input": "0x",
+    "nonce": "0x1",
+    "r": "0x52a6f622013359249316f4c017a67bc2c659f513dac5efea43a84b6ce4e462b1",
+    "s": "0x55ba2a779eaf62efa7d641a32ea329faabf9f097d376e2e400115a5151b9470",
+    "to": "0x658bdf435d810c91414ec09147daa6db62406379",
+    "transactionIndex": "0x0",
+    "type": "0x0",
+    "v": "0xa95",
+    "value": "0x3e8"
+  }
 }
```

### Steps to reproduce

Run hive:
```bash
./hive --client reth --sim ethereum/rpc-compat --sim.limit "eth_getTransactionByHash"
```

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by Rjected** [comment]
blocked by #1905

**Comment by Rjected** [comment]
fixed by #1905
